scio-core/src/main/scala/com/spotify/scio/package.scala

```diff
@@ -30,20 +30,20 @@ import scala.reflect.ClassTag
  */
 package object scio {
 
-  /** [[com.twitter.algebird.Monoid Monoid]] for `Array[Int]`. */
-  implicit val intArrayMon: Monoid[Array[Int]] = new ArrayMonoid[Int]
+  /** [[com.twitter.algebird.Semigroup Semigroup]] for `Array[Int]`. */
```

Comment:
tbh i don't like the fact that this implicits are here with this scope.

Given that they are, this is a breaking change. Is this change really necessary for this PR? Why changing from Monoid to Semigroup?

Reply:
`Array.empty` is not the same as array of fixed size. And there's no way to produce a correct `zeroValue` without seeing an array first.

Removing it from the scope is also a breaking change but probably lesser of 2 evils, assuming not many does `.sum` on arrays.

Reply:
hm, doesn't it only affect the allocation? 

yeah, having this like this leads to that. However, i would probably push this change to another PR.

Reply:
Downstream transform will make assumption about the array dimension, i.e. if you're `sum`ming 40 dimension vectors it'll assume the single output has size 40 and AIOB if it's 0.

Reply:
I think that's somewhat application specific and I it would need a different Monoid instance. This one is a lawful one (as long the array is not mutated) but also the simplest.

I think it will be better to mark these Monoid's as deprecated and then let users defined their instance according to their needs.

Reply:
Empty zero is not a lawful monoid, because  `zero + v` doesn't work if `v` has > 0 dimension and `+` is element-wise plus.